ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx

```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
+    await pickComputeType(container, ComputeType.Standard);
+
+    const enableGpu: HTMLInputElement = screen.getByRole('checkbox', {
+      name: /enable gpus/i,
+    });
+
+    expect(enableGpu.checked).toEqual(false);
+    enableGpu.click();
+    expect(enableGpu.checked).toEqual(true);
+    await pickGpuType(container, 'NVIDIA Tesla T4');
+    await pickGpuNum(container, 2);
+    await pickMainCpu(container, 8);
+    await pickDetachableDiskSize(MIN_DISK_SIZE_GB);
+
+    await clickExpectedButton('Create');
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual('Creating');
+      expect(runtimeApiStub.runtime.gceConfig).toBeUndefined();
+      expect(
+        runtimeApiStub.runtime.gceWithPdConfig.persistentDisk.name
+      ).toEqual('stub-disk');
+      expect(
+        runtimeApiStub.runtime.gceWithPdConfig.gpuConfig.numOfGpus
+      ).toEqual(2);
+    });
+  });
+
+  it('should allow disk deletion when detaching', async () => {
+    setCurrentRuntime(detachableDiskRuntime());
+    setCurrentDisk(existingDisk());
+
+    const { container } = await component();
+    await pickComputeType(container, ComputeType.Dataproc);
+
+    clickExpectedButton('Next');
+    expect(
+      screen.getByText(
+        /your environment currently has a reattachable disk, which will be unused after you apply this update\. /i
+      )
+    ).toBeTruthy();
+
+    togglePDRadioButton();
+
+    clickExpectedButton('Next');
+    clickExpectedButton('Update');
+    await waitFor(async () => {
+      runtimeApiStub.runtime.status = RuntimeStatus.DELETED;
+      expect(runtimeApiStub.runtime.gceWithPdConfig).not.toBeNull();
+    });
+    await waitFor(() => {}, { interval: 1000 });
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual(RuntimeStatus.CREATING);
+      expect(runtimeApiStub.runtime.gceWithPdConfig).toBeUndefined();
+      // expect(runtimeApiStub.runtime.gceWithPdConfig).toBeNull();
+    });
+  });
+
+  it('should allow skipping disk deletion when detaching', async () => {
+    setCurrentRuntime(detachableDiskRuntime());
+    const disk = existingDisk();
+    setCurrentDisk(disk);
+
+    const { container } = await component();
+    await pickComputeType(container, ComputeType.Dataproc);
+
+    clickExpectedButton('Next');
+
+    expect(
+      screen.getByText(
+        /your environment currently has a reattachable disk, which will be unused after you apply this update\./i
+      )
+    ).toBeTruthy();
+
+    // Default option should be NOT to delete.
+    clickExpectedButton('Next');
+    clickExpectedButton('Update');
+    runtimeApiStub.runtime.status = RuntimeStatus.DELETED;
+
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual(RuntimeStatus.CREATING);
+      expect(disksApiStub.disk?.name).toEqual(disk.name);
+    });
+  });
+
+  it('should prevent runtime creation when running cost is too high for free tier', async () => {
```

Comment:
pls rename "free tier" to "initial credits"

Reply:
renamed